src/pages/services/CustumersService.ts

```diff
@@ -0,0 +1,56 @@
+import apiConfig from "@/pages/services/api.ts";
+import {DataResponseWrapper} from "@/pages/services/types/DataResponseWrapper.ts";
+import {CustomerWrapper, SearchCustomerQueryParamsType, SearchCustomerResponseType} from "@/pages/services/types/CustomerType.ts";
+
+export async function searchCustomer(searchQuery: SearchCustomerQueryParamsType): Promise<SearchCustomerResponseType> {
+    const response = await apiConfig
+        .get<DataResponseWrapper<SearchCustomerResponseType>>('/v1/customers', {
+            params: searchQuery
+        });
+
+    return response.data.data;
+}
+
+export async function findCustomer(customerId: string): Promise<CustomerWrapper> {
+    const response = await apiConfig
+        .get<DataResponseWrapper<CustomerWrapper>>(`/v1/customers/${customerId}`);
+
+    return response.data.data;
+}
+
+
+export async function createCustomer(customerData: Partial<CustomerWrapper>): Promise<CustomerWrapper> {
+    const response = await apiConfig
+        .post<DataResponseWrapper<CustomerWrapper>>('/v1/customers', customerData);
+
+    return response.data.data;
+}
```

Comment:
# Primeira sugestão
esse metodo pode receber só o CustomerType, pq se ele receber o CustomerWrapper, quem for chamar vai ter q encapsular o objeto num customer, ele pode passar o customer direto e dentro da função vc encapsula.  
Exemplo de como ta hoje:
```typescript
createCustomer({ customer: { id: "", name: "", cpf: "" } })
```
Se altera o `createCustomer` para receber CustomerType, fica assim:
```typescript
// Atualiza função pra receber um objeto mais simples e faz o wrapper dentro dela
export async function createCustomer(customerData: CustomerType): Promise<CustomerWrapper> {
    const response = await apiConfig
        .post<DataResponseWrapper<CustomerWrapper>>('/v1/customers', { customer: customerData });

    return response.data.data;
}
```
 Agora quem for chamar a função, vai ficar assim:
 ```typescript
 createCustomer({ id: "", name: "", cpf: "" })
 ```
 
# Segunda sugestão
Aqui é bom mesmo usar o Partial? Vc realmente quer deixar todas propriedades opcionais?  
No create, por enquanto, todas propriedades são obrigatórias, ai vc pode usar o CustomerType normal, mas tem uma parada, esse cara tem mapeado o id, mas no create nao precisa de id, então vc pode usar o Omit pra remover o id.  
Ficando assim:
 ```typescript
// Aqui ficou um pouco mais complexo, pq o CertificateType tmb recebe um id, só que pro create, vc nao passa o id no certifcates tmb. 
// Só que como o certificates ta nested (dentro de um obj), tive q remover ele com o primeiro omit e depois readiciona-lo omitindo o id nesse certificate novo 
export async function createCustomer(customerData: Omit<CustomerType, 'id' | 'certificates'> & { certificates: Omit<CertificateType, 'id'>[] }): Promise<CustomerWrapper> {
    const response = await apiConfig
        .post<DataResponseWrapper<CustomerWrapper>>('/v1/customers', { customer: customerData });

    return response.data.data;
}
```